### DISCUSSION TITLE
Handling Server Generated Errors

### DISCUSSION THREAD

**Discussion by steverhoades** [data-driven-forms/react-forms#921] [Need help] [answered]
Is there an example of handling server generated errors someone can point me at? At the moment I'm unable to get my server generated validation errors to display properly. Things like UNIQUE constraint errors etc.

At the moment I am returning the following from the `onSubmit` function:

```
{
   <key>: "error message" 
}
```

According to [React Final Form](https://final-form.org/docs/react-final-form/examples/submission-errors) this should work however I'm not seeing the error state (also no changes present upon inspecting the DOM). Interestingly enough, the targeted field does come into focus, ie. the cursor is placed into the problematic text field.

**Comment by rvsia** [comment]
@steverhoades Hello, thanks for the question.

Do you use any mapper provided by us? I am afraid that we haven't implemented a support for submit errors in them yet, but we could fix it very quickly it if there is a need for it.

**Comment by steverhoades** [comment, marked as answer]
@rvsia Thanks for the quick reply! I am using the `@data-driven-forms/pf4-component-mapper` component mapper. If you can point me in the right direction I will happily assist with that effort.

**Reply by rvsia** [reply]
Awesome!

I have opened an issue about this: https://github.com/data-driven-forms/react-forms/issues/922

If you are interested on working on this feature, just let me know in the comments and I can assign you to the PF4 mapper.

I think it should be pretty straightforward as we need to update two files to support `meta.submitError`:

https://github.com/data-driven-forms/react-forms/blob/f9616f467366f6d98bd435c1eecd80063cb326c9/packages/pf4-component-mapper/src/common/form-group.js

https://github.com/data-driven-forms/react-forms/blob/f9616f467366f6d98bd435c1eecd80063cb326c9/packages/pf4-component-mapper/src/common/show-error.js

+ update all other components that do not use common `FormGroup` (not sure if there is any)
+ add simple test case to https://github.com/data-driven-forms/react-forms/blob/f9616f467366f6d98bd435c1eecd80063cb326c9/packages/pf4-component-mapper/src/tests/form-fields.test.js

And of course, I can help with any of these steps!

**Reply by rvsia** [reply]
As I had some time this morning I prepared a PR to introduce submitErrors to https://github.com/data-driven-forms/react-forms/pull/924

**Reply by rvsia** [reply]
It's released in v2.19.0 :fireworks:

**Reply by steverhoades** [reply]
Wow, that was very fast! I'll give it a shot, thank you!

**Reply by steverhoades** [reply]
Works perfectly!

**Reply by rvsia** [reply]
Thanks for the feedback!
